src/explore-education-statistics-admin/src/pages/release/content/components/ReleaseContent.tsx

```diff
@@ -38,10 +40,23 @@ interface MethodologyLink {
 const ReleaseContent = () => {
   const config = useConfig();
   const location = useLocation();
-
-  const { editingMode } = useEditingContext();
+  const {
+    editingMode,
+    unsavedBlocks,
+    unsavedCommentDeletions,
+  } = useEditingContext();
   const { release } = useReleaseContentState();
   const actions = useReleaseContentActions();
+  const [blockRouteChange, toggleBlockRouteChange] = useToggle(false);
+
+  useEffect(() => {
+    const blocksWithCommentDeletions = Object.entries(unsavedCommentDeletions)
+      .filter(blockWithDeletions => blockWithDeletions[1].length)
+      .map(blockWithDeletions => blockWithDeletions[0]);
+    toggleBlockRouteChange(
+      unsavedBlocks.length > 0 || blocksWithCommentDeletions.length > 0,
+    );
+  }, [toggleBlockRouteChange, unsavedBlocks, unsavedCommentDeletions]);
```

Comment:
Why use state here? Couldn't we `useMemo`?

Reply:
Done